### PR TITLE
Add Kotlin Duration overrides

### PR DESCRIPTION
**Pull request by yschimke**
#fixes https://github.com/square/okhttp/issues/5322

### CODE REVIEW

**Review by JakeWharton — APPROVED** [review]
Why is the annotation for animal sniffer needed?

**Comment by yschimke** [comment]
Because I'm a lazy programmer, who copies and pastes.  will remove.

File: okhttp/src/jvmMain/kotlin/okhttp3/OkHttpClient.kt [hunk 1]
```diff
@@ -1004,6 +1005,21 @@ open class OkHttpClient internal constructor(
       callTimeout(duration.toMillis(), MILLISECONDS)
     }
 
+    /**
+     * Sets the default timeout for complete calls. A value of 0 means no timeout, otherwise values
+     * must be between 1 and [Integer.MAX_VALUE] when converted to milliseconds.
```
Comment (swankjesse) [inline okhttp/src/jvmMain/kotlin/okhttp3/OkHttpClient.kt:1010]: Ick, converting 499 microseconds to no timeout is not good. You’re asking for a very small timeout and we’re giving you none at all.

Comment (swankjesse) [reply okhttp/src/jvmMain/kotlin/okhttp3/OkHttpClient.kt:1010]: checkDuration gets it right. It rejects arguments smaller than 1ms.

https://github.com/square/okhttp/blob/ba3287b7486ec999d4a1960c8bd7eada85e0df3a/okhttp/src/jvmMain/kotlin/okhttp3/internal/-UtilJvm.kt#L100

**Pull request merged**